fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/typing/Type.java

```diff
@@ -381,23 +408,23 @@ private static Type fromProtoType(@Nullable Descriptors.GenericDescriptor descri
         if (protoLabel == FieldDescriptorProto.Label.LABEL_REPEATED) {
             // collection type
             return fromProtoTypeToArray(descriptor, protoType, typeCode, false);
-        } else {
-            if (typeCode.isPrimitive()) {
-                return primitiveType(typeCode, isNullable);
-            } else if (typeCode == TypeCode.ENUM) {
-                final var enumDescriptor = (Descriptors.EnumDescriptor)Objects.requireNonNull(descriptor);
-                return Enum.fromProtoValues(isNullable, enumDescriptor.getValues());
-            } else if (typeCode == TypeCode.RECORD) {
-                Objects.requireNonNull(descriptor);
-                final var messageDescriptor = (Descriptors.Descriptor)descriptor;
-                if (NullableArrayTypeUtils.describesWrappedArray(messageDescriptor)) {
-                    // find TypeCode of array elements
-                    final var elementField = messageDescriptor.findFieldByName(NullableArrayTypeUtils.getRepeatedFieldName());
-                    final var elementTypeCode = TypeCode.fromProtobufType(elementField.getType());
-                    return fromProtoTypeToArray(descriptor, protoType, elementTypeCode, true);
-                } else {
-                    return Record.fromFieldDescriptorsMap(isNullable, Record.toFieldDescriptorMap(messageDescriptor.getFields()));
-                }
+        } else if (typeCode.isPrimitive()) {
+            return primitiveType(typeCode, isNullable);
+        } else if (typeCode == TypeCode.ENUM) {
+            final var enumDescriptor = (Descriptors.EnumDescriptor)Objects.requireNonNull(descriptor);
+            return Enum.fromProtoValues(isNullable, enumDescriptor.getValues());
+        } else if (typeCode == TypeCode.RECORD) {
+            Objects.requireNonNull(descriptor);
+            final var messageDescriptor = (Descriptors.Descriptor)descriptor;
+            if (NullableArrayTypeUtils.describesWrappedArray(messageDescriptor)) {
+                // find TypeCode of array elements
+                final var elementField = messageDescriptor.findFieldByName(NullableArrayTypeUtils.getRepeatedFieldName());
+                final var elementTypeCode = TypeCode.fromProtobufType(elementField.getType());
+                return fromProtoTypeToArray(descriptor, protoType, elementTypeCode, true);
+            } else if (TupleFieldsProto.UUID.getDescriptor().equals(messageDescriptor)) {
```

Comment:
It is a bit complex to understand why the handling of the `UUID` is done here and not in an outer `if`. But after some discussion I understand it better now. Thanks @g31pranjal for the explanation.